xarray/backends/cfgrib_.py

```diff
@@ -0,0 +1,96 @@
+#
+# Copyright 2017-2018 European Centre for Medium-Range Weather Forecasts (ECMWF).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors:
+#   Alessandro Amici - B-Open - https://bopen.eu
+#
+
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+
+from .. import Variable
+from ..core import indexing
+from ..core.utils import Frozen, FrozenOrderedDict
+from .common import AbstractDataStore, BackendArray
+from .file_manager import CachingFileManager
+from .locks import ensure_lock, SerializableLock
+
+# FIXME: Add a dedicated lock just in case, even if ecCodes is supposed to be thread-safe in most
+# circumstances. See: https://confluence.ecmwf.int/display/ECC/Frequently+Asked+Questions
+ECCODES_LOCK = SerializableLock()
+
+
+class CfGribArrayWrapper(BackendArray):
+    def __init__(self, datastore, array):
+        self.datastore = datastore
+        self.shape = array.shape
+        self.dtype = array.dtype
+        self.array = array
```

Comment:
You probably do not want to store a direct reference to `array` here, because it is probably not pickleable, and this object needs to be pickleable to work with Dask distributed.

Reply:
Does `OnDiskArray` store a reference to an open file?

Actually, if the object returned by `cfgrib.open_file` does not include a file object and is already pickleable, then there's no reason why we need to use `CachingFileManager` at all.

Reply:
In fact it doesn't. We `open` the file on every uncached data access. We didn't benchmark this yet so we may end up changing the implementation in the future if we discover that it improve performance. Do you want me to remove the use of `CachingFileManager`?

Reply:
If I'll need it, I'll add an option to keep it open and maintain backward compatibility. No need to be future-proof.